beacon-chain/sync/initial-sync/round_robin.go

```diff
@@ -299,43 +360,100 @@ func validUnprocessed(ctx context.Context, bwb []blocks.BlockWithROBlobs, headSl
 	return bwb[nonProcessedIdx:], nil
 }
 
-func (s *Service) processBatchedBlocks(ctx context.Context, bwb []blocks.BlockWithROBlobs, bFunc batchBlockReceiverFn) (uint64, error) {
-	if len(bwb) == 0 {
+func (s *Service) processBatchedBlocks(ctx context.Context, bwb []blocks.BlockWithROSidecars, bFunc batchBlockReceiverFn) (uint64, error) {
+	bwbCount := uint64(len(bwb))
+	if bwbCount == 0 {
 		return 0, errors.New("0 blocks provided into method")
 	}
+
 	headSlot := s.cfg.Chain.HeadSlot()
-	var err error
-	bwb, err = validUnprocessed(ctx, bwb, headSlot, s.isProcessedBlock)
+	bwb, err := validUnprocessed(ctx, bwb, headSlot, s.isProcessedBlock)
 	if err != nil {
 		return 0, err
 	}
-	if len(bwb) == 0 {
-		return 0, nil
-	}
 
-	first := bwb[0].Block
-	if !s.cfg.Chain.HasBlock(ctx, first.Block().ParentRoot()) {
+	firstBlock := bwb[0].Block
+	if !s.cfg.Chain.HasBlock(ctx, firstBlock.Block().ParentRoot()) {
 		return 0, fmt.Errorf("%w: %#x (in processBatchedBlocks, slot=%d)",
-			errParentDoesNotExist, first.Block().ParentRoot(), first.Block().Slot())
+			errParentDoesNotExist, firstBlock.Block().ParentRoot(), firstBlock.Block().Slot())
 	}
 
-	bv := verification.NewBlobBatchVerifier(s.newBlobVerifier, verification.InitsyncBlobSidecarRequirements)
-	avs := das.NewLazilyPersistentStore(s.cfg.BlobStorage, bv)
-	s.logBatchSyncStatus(first, len(bwb))
-	for _, bb := range bwb {
-		if len(bb.Blobs) == 0 {
+	// Seaerate blocks with blobs from blocks with data columns.
+	fistDataColumnIndex := sort.Search(len(bwb), func(i int) bool {
+		return bwb[i].Block.Version() >= version.Fulu
+	})
+
+	blocksWithBlobs := bwb[:fistDataColumnIndex]
+	blocksWithDataColumns := bwb[fistDataColumnIndex:]
+
+	if err := s.processBlocksWithBlobs(ctx, blocksWithBlobs, bFunc, firstBlock); err != nil {
+		return 0, errors.Wrap(err, "processing blocks with blobs")
+	}
+
+	if err := s.processBlocksWithDataColumns(ctx, blocksWithDataColumns, bFunc, firstBlock); err != nil {
```

Comment:
why wouldn't we need some kind of fulu fork check around thisor with the process blocks with blobs

Reply:
Because the check is done at the very beginning of `s.processBlocksWithBlobs` and `s.processBlocksWithDataColumns`.